Earthfile

```
@@ -51,7 +51,9 @@ test-run:
     RUN ! grep 'Found tool in cache' output
     RUN cat output | grep '^::add-path::' | sed 's/::add-path:://g' > earthly-path
     RUN test "$(cat earthly-path)" = "/root/.earthly/bin"
-    RUN export PATH="$(cat earthly-path):$PATH" && earthly --version | grep '^earthly version v.*linux/amd64; Linux'
+    # [a-zA-Z0-9]* attempt to match a commit hash
+    RUN export PATH="$(cat earthly-path):$PATH" && earthly --version
+    RUN export PATH="$(cat earthly-path):$PATH" && earthly --version | grep '^earthly version v.* [a-zA-Z0-9]* linux/amd64; Alpine Linux'
```

it looks like this regex now requires two spaces for cases like `v0.1.2<space><space>linux/amd64`

should it be something closer to:
```suggestion
    RUN export PATH="$(cat earthly-path):$PATH" && earthly --version | grep '^earthly version v.*( [a-zA-Z0-9]+)? linux/amd64; Alpine Linux'
```

However note that there's already a `.*` should should match everything between `v` and `linux/amd64`, so I'm not sure about this regex change.

What about:

```
RUN export PATH="$(cat earthly-path):$PATH" && earthly --version | tee version.output
RUN grep '^earthly version v.*linux/amd64; Linux' version.output
```

instead?

---

This is very close. 
I've adjusted the regex you've provided just a bit to allow for the specific flavour of Linux as well (Alpine in this example).

```
RUN grep '^earthly version v.*linux/amd64; .* Linux' version.output

```